### PR TITLE
Pass environment to launched processes

### PR DESCRIPTION
**Pull request by mikaelarguedas**
Connects to #37

### CODE REVIEW

File: launch_testing/test/check_env.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+if __name__ == '__main__':
+    import os
```
Comment (dirk-thomas) [inline launch_testing/test/check_env.py]: Please move the import to the module level.

File: launch/launch/launcher.py [hunk 2]
```diff
@@ -342,6 +342,7 @@ def _spawn_process(self, index):
         transport, protocol = yield from loop.subprocess_exec(
             lambda: SubprocessProtocol(p.output_handler),
             *p.cmd,
+            env=p.env,
```
Comment (dirk-thomas) [inline launch/launch/launcher.py]: Can you just add `env` to `kwargs`?

File: launch_testing/test/check_env.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+    assert(os.environ.get('testenv2', 'Not Set') == 'Not Set')
```
Comment (dirk-thomas) [inline launch_testing/test/check_env.py]: Assert is not a function but a keyword. This will always be true since the arguments are a tuple.

Comment (dirk-thomas) [inline launch_testing/test/check_env.py]: This line should better test that the environment variable is not set: `'testenv2' not in os.environ`

**Comment by dirk-thomas** [comment]
The new files need copyright headers.

File: launch/launch/launcher.py (hunk 2 above)
Comment (mikaelarguedas) [inline launch/launch/launcher.py]: Done https://github.com/ros2/launch/pull/38/commits/6368d9c8610db0497b2ce0845fdac263e39a8b82

File: launch_testing/test/check_env.py (hunk 3 above)
Comment (mikaelarguedas) [inline launch_testing/test/check_env.py]: Surprisingly it raised errors as expected with the extra brackets.
Removed in https://github.com/ros2/launch/pull/38/commits/f49d6b681fa3c475976730e42f04b1d1a34de890

File: launch_testing/test/check_env.py (hunk 1 above)
Comment (mikaelarguedas) [inline launch_testing/test/check_env.py]: Done https://github.com/ros2/launch/pull/38/commits/f49d6b681fa3c475976730e42f04b1d1a34de890

File: launch_testing/test/check_env.py (hunk 3 above)
Comment (mikaelarguedas) [inline launch_testing/test/check_env.py]: Makes sense. Changed in https://github.com/ros2/launch/pull/38/commits/ede0634c469bb3a12e3f89a6b2faec93a2be4dfd

**Comment by wjwwood** [comment]
These changes lgtm for merge after the alpha release and after CI.

**Comment by mikaelarguedas** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/1670/)
* OSX [Build Status](http://ci.ros2.org/job/ci_osx/1245/)

**Pull request merged**